File: output/llama_4_maverick/avatar/algo_based_translation/java/python/failed_translation/atcoder_AGC013_A.py

def main():
    N = int(input())
    A = list(map(int, input().split()))
    count = 0
    i = 0
    while i < N:
        if i == N - 1:
            count += 1
            break
        if A[i] == A[i + 1]:
            i += 1
        elif A[i] < A[i + 1]:
            while i < N - 1 and A[i] <= A[i + 1]:
                i += 1
            count += 1
        else:
            while i < N - 1 and A[i] >= A[i + 1]:
                i += 1
            count += 1
    print(count)

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/successful_translation/atcoder_AGC013_A.java

import java.util.*;
import java.io.*;

public class atcoder_AGC013_A {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(input[i]);
        }
        List<Integer> uniqueA = new ArrayList<>();
        for (int i = 0; i < A.length; i++) {
            if (i == 0 || A[i] != A[i - 1]) {
                uniqueA.add(A[i]);
            }
        }
        N = uniqueA.size();
        int i = 1;
        int ans = 0;
        while (i < N - 1) {
            if (uniqueA.get(i - 1) < uniqueA.get(i) && uniqueA.get(i) > uniqueA.get(i + 1) || uniqueA.get(i - 1) > uniqueA.get(i) && uniqueA.get(i) < uniqueA.get(i + 1)) {
                ans++;
                i++;
            }
            i++;
        }
        System.out.println(ans + 1);
    }
}

